jax/_src/xla_bridge.py

```diff
@@ -225,20 +225,37 @@ class BackendRegistration:
   # install a plugin if you don't intend it to be used?
   fail_quietly: bool = False
 
+  # Is this plugin experimental? If a plugin is deemed experimental, we issue
+  # a warning when it is initialized. This is mostly to set user expectations
+  # correctly: we don't want users to think that JAX is buggy because of a
+  # a buggy plugin.
+  experimental: bool = False
```

Comment:
How do we determine that the plugin is experimental ?  @hawkinsp are you using the test result runs by the plugin to determine this ?

Reply:
I made an attempt at criteria in the comment on `_nonexperimental_plugins` below. This is something we will need to evolve over time, I suspect, but the idea here is "warn the user if they might see anything other than success or clear failure with an error message", e.g., wrong outputs. I just don't want users to be surprised.

I didn't want to be too prescriptive about how we evaluate that, but here I was imagining that this would basically be on the honor system: if a plugin author sends a PR adding a plugin to the allowlist and promises that the test suite passes on their hardware (modulo things not implemented), I'd accept that PR.